### DISCUSSION TITLE
Can I create a policy for this? Or is it too complicated?

### DISCUSSION THREAD

**Discussion by cliveportman** [supabase/supabase#8064] [Questions]
I have students in a student table, each containing a reference to a register in the registers table. 
I have teachers in a teachers table, each containing a reference to a school in a school table.
Each register in the registers table contains a reference to the school it's associated with.

For teachers, I fetch their available registers by looking for registers associated with their school.
When the teachers select a register, I need to return all of the students associated with that register, and let the teachers save data for each individual student (probably looping through individually unless there's a way to update multiple at once).

Is there a way to have a policy like this? "If logged-in user's school is the same as the school associated with the register the student is associated with, then let them view and update. Otherwise don't." I figure if I don't have a policy like this, I could potentially be exposing students' data.

Can anyone give me any tips on how they'd approach this? I'd store all of the students' data in the register to simplify things, but student have their own login so they can add their own data.

**Comment by GaryAustin1** [comment]
If you could store the school with each student that seems like it would be a big win for RLS.  Then you only need to select from the teachers table based on their uid() to get their school and check that against the student row school id in the RLS.

Otherwise you need another level between to get the multiple registers (it appears you allow) for the teaches school, and then use that with an IN or ANY test with the student row registers id.
